### PR TITLE
bpo-40465: Deprecate the optional argument to random.shuffle().

### PR DESCRIPTION
**Pull request by rhettinger**
https://bugs.python.org/issue40465

### CODE REVIEW

File: Lib/random.py [hunk 1]
```diff
@@ -321,6 +321,10 @@ def shuffle(self, x, random=None):
                 j = randbelow(i+1)
                 x[i], x[j] = x[j], x[i]
         else:
+            _warn('The *random* parameter to shuffle() deprecated\n'
```
Comment (remilapeyre) [inline Lib/random.py]:
```suggestion
            _warn('The *random* parameter to shuffle() has been deprecated\n'
```

**Review by remilapeyre — COMMENTED** [review]
Hi, I think there is a typo in the warning message.

File: Doc/library/random.rst [hunk 2]
```diff
@@ -208,6 +208,9 @@ Functions for sequences
    generated.  For example, a sequence of length 2080 is the largest that
    can fit within the period of the Mersenne Twister random number generator.
 
+   .. deprecated:: 3.9
+      In the future, the optional argument *random* will be removed.
```
Comment (serhiy-storchaka) [inline Doc/library/random.rst]: It looks like the `shuffle()` function is deprecated. It should be specified that the argument is deprecated.

You can also use the `deprecated-removed` directive to specify the version of removing.

File: Lib/random.py [hunk 3]
```diff
@@ -321,6 +321,10 @@ def shuffle(self, x, random=None):
...
+                  'since Python 3.9 and will be removed in a subsequent '
+                  'version.\n',
```
Comment (serhiy-storchaka) [inline Lib/random.py]:
```suggestion
                  'version.',
```

**Review by serhiy-storchaka — COMMENTED** [review]
Please add an entry in What's New.

**Pull request merged**